internal/services/streamanalytics/stream_analytics_stream_input_eventhub_resource.go

```diff
@@ -135,6 +141,7 @@ func resourceStreamAnalyticsStreamInputEventHubCreateUpdate(d *pluginsdk.Resourc
 				EventHubStreamInputDataSourceProperties: eventHubDataSourceProps,
 			},
 			Serialization: serialization,
+			PartitionKey:  utils.String(d.Get("partition_key").(string)),
```

Comment:
Could we check this before setting? Something like:

```
if v, ok := d.GetOk("partition_key"); ok {
		props.PartitionKey = utils.String(d.Get("partition_key").(string))
	}
```

Reply:
I added an update test, then for completeness I added tests for setting and for clearing the property.
After making this code change, the test for clearing the property started failing. It appears that the API call to Azure is a patch, so if the property is not actually set to nil, it will not get updated. 

So I undid that change and left it how it was, but I also saw that the consumer group was being handled the same way. I wrote a few more tests to verify that the consumer group property had that issue, where once set it is cannot be unset via Terraform(plan will report the change is necessary, but apply will not actually set it).

I put tests and fix for the consumer group here, but did them as a separate commit so I can easily pull that out and put it against another issue specific to that problem if that is preferred.

Reply:
Thanks for the update @darren-clark. It would be good to combine these tests into the main update test and add the removal test as an additional step at the end of that.

Reply:
Consolidated the update tests. Update tests now  
1) Creates with consumer group and partition key
2) Updates with new values for consumer group, partition key, and event hub.
3) Updates with removal of consumer group and partition key

Added one additional test to create with no consumer group or partition key. i.e. Create a resource with no optional properties.